benchmarks/src/jmh/java/software/amazon/jdbc/benchmarks/ReadWriteSplittingLoadBenchmarks.java

```diff
@@ -0,0 +1,197 @@
+    props.setProperty(PGProperty.SOCKET_TIMEOUT.getName(), "5");
+
+    return props;
+  }
+
+  protected static Properties initReadWritePluginProps() {
+    final Properties props = initNoPluginPropsWithTimeouts();
+    props.setProperty(PropertyDefinition.PLUGINS.name, "auroraHostList,readWriteSplitting");
+    return props;
+  }
+
+  protected static Properties initReadWritePluginLoadBalancingProps() {
+    final Properties props = initNoPluginPropsWithTimeouts();
+    props.setProperty(PropertyDefinition.PLUGINS.name, "auroraHostList,readWriteSplitting");
+    props.setProperty(ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.name, "true");
+    return props;
+  }
+
+  protected static Connection connectToInstance(String instanceUrl, Properties props)
+      throws SQLException {
+    return DriverManager.getConnection(instanceUrl, props);
+  }
+
+  @Benchmark
+  public void noPluginEnabledBenchmarkTest() {
+    final List<Thread> connectionThreadsList = new ArrayList<>(NUM_THREADS);
+
+    for (int i = 0; i < NUM_THREADS; i++) {
+      connectionThreadsList.add(getThread_PGReadWriteSplitting(initNoPluginPropsWithTimeouts()));
+    }
+
+    for (Thread connectionThread : connectionThreadsList) {
+      connectionThread.start();
+    }
+
+    for (Thread connectionThread : connectionThreadsList){
+      connectionThread.interrupt();
+    }
+  }
+
+  @Benchmark
+  public void readWriteSplittingPluginEnabledBenchmarkTest() {
+    final List<Thread> connectionThreadsList = new ArrayList<>(NUM_THREADS);
+
+    for (int i = 0; i < NUM_THREADS; i++) {
+      connectionThreadsList.add(getThread_PGReadWriteSplitting(initReadWritePluginProps()));
+    }
+
+    for (Thread connectionThread : connectionThreadsList) {
+      connectionThread.start();
+    }
+
+    for (Thread connectionThread : connectionThreadsList){
+      connectionThread.interrupt();
+    }
+  }
+
+  @Benchmark
+  public void readWriteSplittingPluginLoadBalancingEnabledBenchmarkTest() {
+    final List<Thread> connectionThreadsList = new ArrayList<>(NUM_THREADS);
+
+    for (int i = 0; i < NUM_THREADS; i++) {
+      connectionThreadsList.add(getThread_PGReadWriteSplitting(initReadWritePluginLoadBalancingProps()));
+    }
+
+    for (Thread connectionThread : connectionThreadsList) {
+      connectionThread.start();
+    }
+
+    for (Thread connectionThread : connectionThreadsList){
+      connectionThread.interrupt();
+    }
+  }
+
+  private Thread getThread_PGReadWriteSplitting(Properties props) {
+    return new Thread(() -> {
+      try {
+
+        Connection conn = connectToInstance(POSTGRESQL_CONNECTION_STRING, props);
+
+        Thread.sleep(5000);
+
+        // execute long query
+        final Statement statement = conn.createStatement();
+        statement.executeQuery(QUERY_1);
+
+        try (final ResultSet result = statement.executeQuery(QUERY_1)) {
+          fail("Sleep query finished, should not be possible with the network down.");
```

Comment:
I'm a bit confused about this line. Where are we bringing the network down, and why? Why do we not expect the query to complete? Why does it say the "sleep query" finished - it seems like we are not executing a sleep statement but instead the query defined by QUERY_1?